pyccel/codegen/printing/ccode.py

```diff
@@ -267,6 +267,7 @@ def __init__(self, parser, settings=None):
         self._additional_imports = set(['stdlib'])
         self._parser = parser
         self._additional_code = ''
+        self._unused_variable = None
```

Comment:
Can you explain why there can only be one unused variable at a time? Are we sure that this won't create problems in the case of functions within functions? Or is the placement of the return statement sufficient to avoid problems here?

Reply:
you are right about functions within functions, So we can use a list.

Reply:
😅 but C does not support nested functions

Reply:
This is more to do with the processing order than the syntax. If we did:
```
def _visit_FunctionDef(self, expr):
    nested_funcs = self._visit(expr.funcs)
    ... # Existing code
    return code + '\n' + nested_funcs
```
then we could still potentially translate nested functions. But the `return` always has to be the last statement in a function so if this object is only used for returns (see rename discussion below) then I don't think there is a problem